### ISSUE TITLE
Combat resolve state is not cleared properly between game sessions

### ISSUE THREAD

**Issue by pileofwolves** [pileofwolves/finmer#2]
When exiting a game session and loading another game (going through the main menu) does not properly reset all state in the GameUI class. This will lead to a bunch of unexpected things (widgets, animations) showing up when loading into the new game.

This should be relatively easy to fix by adding a few missing cleanup calls in `GameUI.Reset()`.

**Comment by Wh0Kn0w5** [comment]
Ok, as it currently stands, there are just the ClearButtons and ClearLog methods in GameUI.Reset(), which presumably clears the buttons and messages on the screen when exiting the game.

As far as additional cleanup calls, what else would need to be reset/removed during the proper shutdown of the application?

**Comment by pileofwolves** [comment]
> 
> 
> Ok, as it currently stands, there are just the ClearButtons and ClearLog methods in GameUI.Reset(), which presumably clears the buttons and messages on the screen when exiting the game.
> 
> As far as additional cleanup calls, what else would need to be reset/removed during the proper shutdown of the application?

All fields of `GameUI` should be reset to defaults. I believe the only one that's missing is `GameUI.CombatStateViewModel`, but I'm not 100% sure.

You can test your fix by killing the player in combat, such as with this console command (click the Script Console button in-game):
```lua
local c = Combat2(); local r = Creature("CR_MQ02_Goon1"); r.Strength = 20; c:AddParticipant(Player); c:AddParticipant(r); c:Begin()
```
then returning to the main menu and starting a new game. The combat panel from the previous game will still be visible.

**Comment by Wh0Kn0w5** [comment]
Seems calling the CloseCombatUI() Method from the Combat Display script resets the combat menu so that the compass, clock and player stats properly show.

Do you know of any problems calling that method might cause?

**Comment by pileofwolves** [comment]
`CloseCombatUI()` is used by the combat system (`SceneCombat2`) to animate the combat side panel flying out. It wouldn't be appropriate to call from `GameUI.Reset()` directly since it is intended to wait on an animation which is not playing at all (and will freeze the main thread in doing so).

**Comment by Wh0Kn0w5** [comment]
So then I guess I need to ask for the best way to handle this issue. 

If the goal is to properly close the combat menu without causing thread conflicts further on down the line, would it be best then to simulate combat ending normally by calling the same scripts as in the SceneCombat2 system?

**Comment by pileofwolves** [comment]
I think it could be solved by either just setting `CombatStateViewModel` to null in `Reset()`, or by ditching and re-creating the whole `GameUI.Instance`. The former would be a slightly simpler temp fix, but now that I'm thinking about it, the latter would help prevent this type of issue in the future entirely.

**Comment by Wh0Kn0w5** [comment]
Ok, I've tried both of those methods to the best of my ability.

Setting the CombatStateViewModel to null had some unexpected result when launching it, so I opted not to go that route. (See attached screenshot for what I encountered upon starting a new game.)
![sBND2zJSsP](https://user-images.githubusercontent.com/89660688/132116572-48702b23-9362-4d65-8fdf-0c9931481dd3.png)

I believe that I have created a new Instance of GameUI, but I am not entirely confident in my implementation method. I'll make another pull request off my work branch and you'll be able to check me on my work.

**Comment by pileofwolves** [comment]
Just for posterity: the odd result you got with the first approach (going by the screenshot) appears to be because `IsInCombat` also needed to be reset (setting it to false would close the dice tracker and the empty combatant panel on the left).

However, the other approach (which is merged by now, cheers!) is better all around and will prevent this class of issues entirely 😄

**Comment by Wh0Kn0w5** [comment]
cool. Now just need to update the Trello board to put this issue to bed.
